test/unit/orm-adapters-memory-test.js

```diff
@@ -0,0 +1,14 @@
+import test from 'ava';
+import { MemoryAdapter } from '../../lib/data/orm-adapters/memory';
```

Comment:
Yea, so @knownasilya is correct on the lack of curly braces (the MemoryAdapter class is the default export from that file, so no curlies are needed).

You can import from `'denali'`, yes. The reason that works is actually the same reason that the `lib` import isn't working too.

Remember that the CLI builds your app before running the tests, so the `test/unit/foo.js` file isn't actually executing - `dist/test/unit/foo.js` is. So if your import paths aren't working as expected, make sure you check the contents of `dist/` when trying to debug.

Now for why your `lib` import doesn't work, and `denali` imports do.

You can see why by exploring the `dist/` folder after running `denali test`. Because Denali is packaged as an addon, when you "build" it, you're actually building the dummy app which automatically includes the addon it's inside. You can see that there's a `dist/app/` and `dist/config/` folder that match the `test/dummy/` contents. 

The actual source for an addon under test is actually compiled into `dist/node_modules/my-addon-name`, i.e. as a dependency of the dummy app. Normally, this would result in your addon's test files ending up inside `dist/node_modules/my-addon/test`, and they would get missed. To correct that, when you run `denali test` in an addon the CLI pulls a minor of sleight of hand and yanks the addon's test up into the built dummy app's tests (i.e. `my-addon/test` ends up in `my-addon/dist/test`, not `my-addon/dist/node_modules/my-addon/test`).

Thus, from your tests, you can directly import the addon as if it was a regular node_modules dependency (because it is in the compiled version!), but you can't import the relative lib path, because that would try to walk up into the dummy app directory, which doesn't have a `lib` folder.

Hopefully that makes sense. I'm not in love with the whole "the compiled version has a different structure" but this kind of sleight of hand is only really present for addons (a compiled app should be close enough in structure to the source that you'd have to try pretty hard to find discrepancies in behavior). And since addon authoring is likely only to be tackled by the more experienced users, I'm hoping the nuances won't be overwhelming by then (with proper docs).